roles/slurm/tasks/build.yml

```diff
@@ -110,6 +110,13 @@
   when: slurm_build
   block:
 
+    # This service should be stopped before uninstalling slurm
+  - name: stop docker slurm exporter service to prevent subfolder creation
+    service:
+      name: "docker.slurm-exporter.service"
+      state: stopped
+    when: slurm_build and (ansible_distribution == 'Ubuntu') and ("{{ansible_hostname}}" not in groups['slurm-node'])
```

Comment:
This fails on initial installs because the exporter hasn't been installed yet. The error looks like this:

```
fatal: [virtual-login01-2]: FAILED! => changed=false 
  invocation:
    module_args:
      daemon_reexec: false
      daemon_reload: false
      enabled: null
      force: null
      masked: null
      name: docker.slurm-exporter.service
      no_block: false
      scope: system
      state: stopped
  msg: 'Could not find the requested service docker.slurm-exporter.service: host'
```

You should ensure that service facts are being collected, then add a condition so this only executes when the service exists. I.e., something like this:

```
- name: gather service facts
  ansible.builtin.service_facts:

- name: stop docker slurm exporter service to prevent subfolder creation
  service:
      name: "docker.slurm-exporter.service"
      state: stopped
  when: 
  - slurm_build
  - ansible_distribution == 'Ubuntu'
  - "{{ansible_hostname}}" not in groups['slurm-node']
  - "docker.slurm-exporter.service" in ansible_facts.services
```

(I haven't tested this change directly, but I'm pretty sure the syntax looks like that!)

Reply:
@ajdecon  I incorporated your suggestion and also restarted the service back again in the end. Ran a test deployment and also rebased to master.

Reply:
By the way, may be because of older ansible version, the `when` list doesn't work so I use the explicit logical `and`.